src/Model/Record.php

```diff
@@ -131,18 +131,18 @@ final class Record
      *   considered a BC break. See the format definition of known types above
      *   for more details.
      *
-     * @var string|string[]|array
+     * @var string|array<string>
      */
     public $data;
 
     /**
-     * @param string                $name
-     * @param int                   $type
-     * @param int                   $class
-     * @param int                   $ttl
-     * @param string|string[]|array $data
+     * @param string               $name
+     * @param int                  $type
+     * @param int                  $class
+     * @param int                  $ttl
+     * @param string|array<string> $data
      */
-    public function __construct($name, $type, $class, $ttl, $data)
+    public function __construct(string $name, int $type, int $class, int $ttl, $data)
```

Comment:
Not sure if this is a valid suggestion, more of a question if we should add this as well :D

```suggestion
    public function __construct(string $name, int $type, int $class, int $ttl, string|array $data)
```

Reply:
Union types are PHP 8.0+, so yes would love to add this, just can't since we only support 7.1+ for v3

Reply:
Ah okay :+1:

Reply:
Don't get me wrong, I would love to put it in. But this is one for v4 :wink: